draft-ietf-mmusic-dtls-sdp.xml

```diff
@@ -368,10 +386,12 @@
             <t>
                 If the offerer inserts the SDP 'setup' attribute with an 'actpass' or 'passive' attribute value, the
                 offerer MUST be prepared to receive a DTLS ClientHello message (if a new DTLS association
-                is established by the answerer) from the answerer before the offerer receives the SDP answer.
+                is established by the answerer) from the answerer before the offerer receives the SDP answer. In such
+                case the offerer MAY cache the ClientHello message, but the offerer MUST NOT initiate a DTLS

```

Comment:
I'm concerned that this is too restrictive. Even if completing the DTLS handshake before the answer is received is impossible with ICE, why isn't it possible with plain UDP, or some other arbitrary datagram-based protocol?

If this change **is** made, there are some other places in this document that need updating. For example, Section 5.1:

> Note that it is permissible to wait until the other side's fingerprint(s) has been received before establishing the connection; however, this may have undesirable latency effects.

Also, Section 10.2.1 (updating RFC 5763):

> The offerer may only tentatively accept the answerer's certificate since it may not yet have the answerer's certificate fingerprint.